meshio/gmsh/_gmsh41.py

```diff
@@ -216,31 +216,15 @@ def _read_elements(f, point_tags, physical_tags, is_ascii, data_size):
         (physical_tag, tpe, inv_tags[d[:, 1:] - 1]) for physical_tag, tpe, d in data
     ]
 
-    cells = {}
+    cells = []
     for physical_tag, key, values in data:
-        if key in cells:
-            cells[key] = numpy.concatenate([cells[key], values])
-            if physical_tag:
-                if "gmsh:physical" not in cell_data:
-                    cell_data["gmsh:physical"] = {}
-                if key not in cell_data["gmsh:physical"]:
-                    cell_data["gmsh:physical"][key] = {}
-                cell_data["gmsh:physical"][key] = numpy.concatenate(
-                    [
-                        cell_data["gmsh:physical"][key],
-                        physical_tag[0] * numpy.ones(len(values), int),
-                    ]
-                )
-        else:
-            cells[key] = values
-            if physical_tag:
-                if "gmsh:physical" not in cell_data:
-                    cell_data["gmsh:physical"] = {}
-                if key not in cell_data["gmsh:physical"]:
-                    cell_data["gmsh:physical"][key] = {}
-                cell_data["gmsh:physical"][key] = physical_tag[0] * numpy.ones(
-                    len(values), int
-                )
+        cells.append((key, values))
+        if physical_tag:
+            if "gmsh:physical" not in cell_data:
+                cell_data["gmsh:physical"] = {}
+            cell_data["gmsh:physical"].append(
+                (key, physical_tag[0] * numpy.ones(len(values), int))
```

Comment:
@nschloe Why you do `(key, tags)` in here? Using `tags` would be just fine, the cell type in here is redundant, and hardly useful if you have many blocks with the same cell type. The cell data ordering is implicit from the `mesh.cells` attribute. This broke compatibility with the gmsh 2.2 reader/writer, particularly if you want to read v4.1 and dump v2.2 (which I was using routinely before these changes). I'm working on fixing this issue. 

Would you accept this change (plus other stuff to make conversion from 4.1 to 2.2)?

```diff
--- a/meshio/gmsh/_gmsh41.py
+++ b/meshio/gmsh/_gmsh41.py
@@ -222,7 +222,7 @@ def _read_elements(f, point_tags, physical_tags, is_ascii, data_size):
             if "gmsh:physical" not in cell_data:
                 cell_data["gmsh:physical"] = []
             cell_data["gmsh:physical"].append(
-                (key, physical_tag[0] * numpy.ones(len(values), int))
+                physical_tag[0] * numpy.ones(len(values), int)
             )
```

Reply:
>  Using tags would be just fine

Yep, that's an oversight. PR welcome!